### PR TITLE
API: /details-multiple - returns multiple account details

### PR DESCRIPTION
**Pull request by AfuSensi**
Tested and works locally, if anyone sees a problem with this, please let
me know.

### CODE REVIEW

**Comment by GingerAdonis** [comment]
I'll check this today

File: routes/api/account.js [hunk 1]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+            log.error(error);
+            continue;
+        }
+
+        if (!user) {
+            continue;
+        }
+
+        let singleUser = [{
+            users: user.getId(),
```
Comment (AfuSensi) [inline routes/api/account.js:150]: users should be userId

File: routes/api/account.js [hunk 2]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+        } catch (error) {
+            log.error(error);
+            continue;
+        }
+
+        if (!user) {
+            continue;
+        }
+
+        let singleUser = [{
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
        const singleUser = {
```

File: routes/api/account.js [hunk 3]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
             title: undefined //Deprecated
         }
     });
+});
+
+router.post('/details-multiple', async (req, res) => {
+    const usersArray = req.body.users;
+    if (typeof usersArray != 'object') {
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
    if (typeof(usersArray) !== 'object') {
```

File: routes/api/account.js [hunk 4]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+            emailAddress: undefined, //Deprecated
+            joinDate: user.getCreated(),
+            joinTimestamp: user.getCreated() ? Math.round(user.getCreated().getTime() / 1000) : undefined,
+            coinsBalance: user.getCoins().getBalance(),
+            profile: {
+                photo: user.getAvatar(),
+                photoThumb: user.getAvatarThumb(),
+                title: undefined //Deprecated
+            }
+        }];
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
        };
```

File: routes/api/account.js [hunk 5]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+        res.json({
+            error: 1,
+            errorMessage: 'Invalid users array'
+        });
+        return;
+    }
+
+
+    let userReturn = [];
+    for (let i = 0; i < usersArray.length; ++i) {
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
    for (const requestedUser of usersArray) {
```

File: routes/api/account.js [hunk 6]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+        let userId = parseInt(usersArray[i].userId, 10);
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
        const userId = parseInt(requestedUser.userId, 10);
```

File: routes/api/account.js [hunk 7]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+            coinsBalance: user.getCoins().getBalance(),
+            profile: {
+                photo: user.getAvatar(),
+                photoThumb: user.getAvatarThumb(),
+                title: undefined //Deprecated
+            }
+        }];
+        userReturn.push(singleUser);
+    }
+    res.json({userReturn});
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
    res.json({users: userReturn});
```
Better API output syntax

**Review by GingerAdonis — CHANGES_REQUESTED** [review]
Several optimizations and bug fixes

File: routes/api/account.js [hunk 8]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+    if (usersArray.length < 1 ) {
+        res.json({
+            error: 1,
+            errorMessage: 'Invalid users array'
+        });
+        return;
+    }
+
+
+    let userReturn = [];
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
    const userReturn = [];
```

File: routes/api/account.js [hunk 9]
```diff
@@ -103,4 +103,63 @@ router.post('/details', async (req, res) => {
...
+    for (const requestedUser of usersArray) {
+        const userId = parseInt(requestedUser.userId, 10);
+        
+        if (isNaN(userId)) {
+            console.log('userID NaN');
```
Comment (GingerAdonis) [inline routes/api/account.js]:
```suggestion
            log.warn('userID NaN');
```

**Review by GingerAdonis — CHANGES_REQUESTED** [review]

**Comment by GingerAdonis** [comment]
2 minor changes needed. Apart from that it looks good to me.

**Comment by AfuSensi** [comment]
Thank you @JarnoVgr , what changes are still needed?

**Pull request merged**